client/Bootstrap/Navigation/useRouteConfig/useRouteConfig.hook.ts

```diff
@@ -18,44 +22,61 @@ const removeWhitespace = (name) => name.replace(/\s/g, '');
 
 const removePathVars = (name) => name.replace(/\${[^}]*\}/g, '');
 
-const contextReducer = (acc, context, page, leaves) => {
-  // TRANSLATE THIS ONCE 188N IS IN
-  const translatedName = context.name;
+const checkFeatureFlag = (page, context, featureFlags) => {
+  if (get(featureFlags.client, context.feature_flag)) {
+    return page.contentComponent;
+  } else {
+    return Error404;
+  }
+};
+
+const contextReducer = (
+  acc,
+  context,
+  page,
+  leaves,
+  featureFlags,
+  translate
+) => {
+  const translatedName = translate(context.name);
   const linkKey = generateKey(translatedName, 'link', context.properties);
   const routeKey = generateKey(translatedName, 'route', context.properties);
+  const component = checkFeatureFlag(page, context, featureFlags);
 
-  if (isTopLevelNav(context.path)) {
-    acc.links.push({
-      to: context.path,
-      key: linkKey,
-      children: translatedName,
-    });
-  } else {
-    leaves.push({
-      parent: `/${extractTopLevelPath(context.path)}`,
-      child: {
-        path: context.path,
-        name: translatedName,
-      },
-    });
+  if (component === page.contentComponent) {
+    if (isTopLevelNav(context.path)) {
+      acc.links.push({
+        to: context.path,
+        key: linkKey,
+        children: translatedName,
+      });
+    } else {
+      leaves.push({
+        parent: `/${extractTopLevelPath(context.path)}`,
+        child: {
+          path: context.path,
+          name: translatedName,
+        },
+      });
+    }
+
+    acc.meta[context.path] = {
+      name: translatedName,
+      pageType: context.pageType,
+      order: context.order,
+      properties: context.properties ? context.properties : {},
+      isTopLevel: isTopLevelNav(context.path),
+      icon: context.icon,
+      leaves: [],
+    };
   }
 
   acc.routes.push({
     path: context.path,
     key: routeKey,
-    componentForRoute: page.contentComponent,
+    componentForRoute: component,
```

Comment:
```suggestion
    componentForRoute: pageEnabled ? page.component : Error404,
```

Reply:
Solid suggestion, will definitely do this